vortex-serde/src/layouts/read/builder.rs

```diff
@@ -85,21 +87,32 @@ impl<R: VortexReadAt> LayoutBatchStreamBuilder<R> {
             RelativeLayoutCache::new(message_cache.clone(), footer_dtype.clone()),
         )?;
 
-        let filter_reader = self
-            .row_filter
-            .as_ref()
-            .map(|filter| {
-                footer.layout(
-                    Scan::new(Some(Arc::new(filter.clone()))),
+        let row_filter_and_reader = match self.row_filter {
+            None => None,
+            Some(row_filter) => {
+                let mut references = HashSet::new();
+                row_filter.collect_references(&mut references);
+                let select_filtering_columns =
+                    Select::Include(references.into_iter().map(|x| x.clone()).collect());
+                let layout = footer.layout(
+                    Scan::new(Some(Arc::new(select_filtering_columns))),
```

Comment:
you still want to pass the filter here?

Reply:
Yeah, this is the only way I know of to pushdown a column projection to the layout.

Reply:
We will use references of the filter as the projection (that logic lives in column layout)